ocular/src/chain/client/cache.rs

```diff
@@ -0,0 +1,528 @@
+use crate::error::CacheError;
+
+use serde::{Deserialize, Serialize};
+use std::collections::HashSet;
+use std::fs::File;
+use std::os::unix::fs::PermissionsExt;
+
+// Constants
+const DEFAULT_FILE_CACHE_DIR: &str = "/.ocular/cache";
```

Comment:
The '~' prefix gets added later in the code flow. For some reason '~' doesn't get recognized as the home dir so we have to build it by retrieving the home dir instead of using the shorthand.